src/web3/store/walletSlice.ts

```diff
@@ -69,27 +71,31 @@ export function createWeb3Slice({
 
       const impersonatedAddress = get()._impersonatedAddress;
       set({ walletActivating: true });
+      set({ walletConnectionError: '' });
       const connector = get().connectors.find(
-        connector => getConnectorName(connector) === walletType
+        (connector) => getConnectorName(connector) === walletType
       );
-      if (connector) {
-        switch (walletType) {
-          case "Impersonated":
-            if (impersonatedAddress) {
-              await connector.activate(impersonatedAddress);
-            }
-            break;
-          case "Coinbase":
-          case "Metamask":
-            await connector.activate(
-              getAddChainParameters(desiredChainID)
-            );
-            break;
-          case "WalletConnect":
-            await connector.activate(desiredChainID);
-            break;
+      try {
+        if (connector) {
+          switch (walletType) {
+            case 'Impersonated':
+              if (impersonatedAddress) {
+                await connector.activate(impersonatedAddress);
+              }
+              break;
+            case 'Coinbase':
+            case 'Metamask':
+              await connector.activate(getAddChainParameters(desiredChainID));
+              break;
+            case 'WalletConnect':
+              await connector.activate(desiredChainID);
+              break;
+          }
+          setLocalStorageWallet(walletType);
         }
-        setLocalStorageWallet(walletType);
+      } catch (e: any) {
```

Comment:
```ts
if (e instanceof Error) {
  set({ walletConnectionError: e.message ? e.message.toString() : e.toString() });
}
```
may be let's do like so instead of any?